internal/future/statestore/redis.go

```diff
@@ -94,29 +96,208 @@ func redisCloser(closer func() error) {
 	}
 }
 
-// CreateTicket creates a new Ticket in the state storage. This method fails if the Ticket already exists.
-func (rb *redisBackend) CreateTicket(context.Context, pb.Ticket, int) error {
-	return status.Error(codes.Unimplemented, "not implemented")
+// CreateTicket creates a new Ticket in the state storage. If the id already exists, it will be overwritten.
+func (rb *redisBackend) CreateTicket(ctx context.Context, ticket *pb.Ticket, ttl int) error {
+	redisConn, err := rb.redisPool.GetContext(ctx)
+	if err != nil {
+		redisLogger.WithFields(logrus.Fields{
+			"error": err.Error(),
+		}).Error("failed to connect to redis")
+		return err
```

Comment:
Prefer returning gRPC error codes here because at this point we know if the error is retryable or not.

Reply:
Done.